From 5695eab02bb88318bf99d568b6d0e4592d5f9981 Mon Sep 17 00:00:00 2001
From: Karthik Ramanan <a0393906@ti.com>
Date: Mon, 12 Mar 2018 10:47:02 +0530
Subject: [PATCH 2/4] weston: Allow visual_id to be 0

The inquiry of visual id from egl API eglGetConfigAttrib(EGL_NATIVE_VISUAL_ID)
is an optional feature. The visual id will be set to 0 if this feature is
not supported. Therefore, the return condition @function match_config_to_visual()
should be (id == visual_id || id == 0) instead of (id == visual_id)

Upstream status: Pending

Signed-off-by: Eric Ruei <e-ruei1@ti.com>
Signed-off-by: Karthik Ramanan <a0393906@ti.com>
---
 libweston/gl-renderer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libweston/gl-renderer.c b/libweston/gl-renderer.c
index c6091af..9299dc6 100644
--- a/libweston/gl-renderer.c
+++ b/libweston/gl-renderer.c
@@ -2575,7 +2575,7 @@ match_config_to_visual(EGLDisplay egl_display,
 				&id))
 			continue;
 
-		if (id == visual_id)
+		if (id == visual_id || id == 0)
 			return i;
 	}
 
-- 
1.9.1

